File: scripts/plotly/example_1.js
import Plotly from 'plotly.js-dist-min';

// Time series with range slider
const timeSeriesData = [{
  x: timestamps,
  y: values,
  type: 'scatter',
  mode: 'lines',
  name: 'Revenue',
  line: { color: '#4CAF50', width: 2 }
}];

const timeSeriesLayout = {
  title: 'Revenue Over Time',
  xaxis: {
    rangeslider: { visible: true },
    type: 'date'
  },
  yaxis: { title: 'Revenue ($)' },
  template: 'plotly_dark'
};

Plotly.newPlot('timeseries', timeSeriesData, timeSeriesLayout);

// 3D scatter plot with WebGL
const scatter3dData = [{
  x: xData,
  y: yData,
  z: zData,
  mode: 'markers',
  type: 'scatter3d',
  marker: {
    size: 5,
    color: colorValues,
    colorscale: 'Viridis',
    opacity: 0.8
  }
}];

Plotly.newPlot('scatter3d', scatter3dData, {
  scene: {
    xaxis: { title: 'X Axis' },
    yaxis: { title: 'Y Axis' },
    zaxis: { title: 'Z Axis' }
  }
});

// Real-time streaming update
function updateChart(newPoint) {
  Plotly.extendTraces('timeseries', {
    x: [[newPoint.timestamp]],
    y: [[newPoint.value]]
  }, [0]);
}

File: scripts/plotly/example_1.py
import plotly.express as px
import pandas as pd

# Load data
df = pd.read_parquet('sales_data.parquet')

# Interactive scatter with trendline
fig = px.scatter(
    df,
    x='revenue',
    y='profit',
    color='category',
    size='volume',
    hover_data=['product_name'],
    trendline='ols',
    title='Revenue vs Profit by Category'
)

fig.update_layout(template='plotly_dark')
fig.show()

# Animated time series
fig = px.line(
    df,
    x='date',
    y='value',
    color='region',
    animation_frame='year',
    range_y=[0, df['value'].max() * 1.1]
)
fig.show()